packages/core/src/config/config.ts

```diff
@@ -67,36 +67,47 @@ export type ContractRequired<
   TNetworkNames extends string,
   TAbi extends Abi | unknown,
   TEventName extends string,
+  TFactoryEvent extends AbiEvent | undefined,
 > = {
   /** Contract application byte interface. */
-  abi: TAbi;
+  abi: Abi;
   /**
    * Network that this contract is deployed to. Must match a network name in `networks`.
    * Any filter information overrides the values in the higher level "contracts" property.
    * Factories cannot override an address and vice versa.
    */
   network:
-    | Partial<Record<TNetworkNames, Partial<ContractFilter<TAbi, TEventName>>>>
+    | Partial<
+        Record<
+          TNetworkNames,
+          Partial<ContractFilter<TAbi, TEventName, TFactoryEvent>>
+        >
+      >
     | TNetworkNames;
 };
 
 /** Fields for a contract used to filter down which events indexed. */
 export type ContractFilter<
   TAbi extends Abi | unknown,
   TEventName extends string,
+  TFactoryEvent extends AbiEvent | undefined,
 > = (
   | {
       address?: `0x${string}` | readonly `0x${string}`[];
+      factory?: never;
     }
   | {
+      address?: never;
       /** Factory contract configuration. */
-      factory: {
+      factory?: {
```

Comment:
Interesting - were we previously requiring a `factory` field if `address` was not defined?

Reply:
No because address was also optional. I tested this a bunch and this is the setup we want.